lib/travis/api/app/endpoint/assembla.rb

```diff
@@ -49,7 +67,12 @@ def validate_request!
       end
 
       def check_required_fields
-        missing = REQUIRED_JWT_FIELDS.select { |f| @jwt_payload[f].nil? || @jwt_payload[f].to_s.strip.empty? }
+        required_fields = if request.path_info.end_with?('/notify')
+                           REQUIRED_NOTIFY_FIELDS
+                         else
+                           REQUIRED_JWT_FIELDS
+                         end
```

Comment:
can't we write it in one sentence like?

`required_fields = request.path_info.end_with?('/notify') ? REQUIRED_NOTIFY_FIELDS : REQUIRED_JWT_FIELDS`